storyruntime/Story.py

```diff
@@ -49,6 +49,44 @@ def new_frame(self, line_number: str):
     def get_stack(self) -> []:
         return self._stack
 
+    @contextmanager
+    def new_context(self):
+        """
+        Creates a new context in the stack
+        """
+        self._context.append({})
+        yield
+        self._context.pop()
+
+    def global_context(self):
+        """
+        Returns the global context for the story
+        """
+        return self.app.story_global_context.setdefault(self.name, {})
+
+    def resolve_context(self, variable):
+        """
+        Used by set_variable to determine the context for a given variable
+        """
+        global_context = self.global_context()
+        for ctx in chain(reversed(self._context), (global_context,)):
+            if variable in ctx:
+                return ctx
+        # variable not found in context
+        return self._context[-1] if len(self._context) > 0 else global_context
+
+    def get_context(self):
```

Comment:
As mentioned before, this isn't a getter. It does a lot more work that that. It should be named accordingly, e.g.

```suggestion
    def build_combined_context(self):
```